weld/codegen/llvm2/mod.rs

```diff
@@ -1245,8 +1271,60 @@ impl LlvmGenerator {
                     unreachable!()
                 }
             }
-            Sort { .. } => {
-                unimplemented!() // Sort
+            Sort { ref child, ref cmpfunc } => { // cmpfunc is an SirFunction
+                let output_pointer = context.get_value(output)?;
+                let output_type = context.sir_function.symbol_type(
+                    statement.output.as_ref().unwrap())?;
+                
+                let ref cmpfunc_func = context.sir_program.funcs[*cmpfunc];
+                if let Vector(ref elem_ty) = *output_type {
+                    let child_value = self.load(context.builder, context.get_value(child)?)?;
+                    let child_type = context.sir_function.symbol_type(child)?;
+
+                    use ast::Type::Scalar;
+                    use ast::ScalarKind;
+                    use self::vector::VectorExt;
+                    let zero = self.zero(self.i64_type());
+                    let child_elems = self.gen_at(context.builder, child_type, child_value, zero)?;
+                    let elems_ptr = LLVMBuildBitCast(context.builder, child_elems,
+                                                     self.void_pointer_type(),
+                                                     c_str!(""));
+                    let size = self.gen_size(context.builder, child_type, child_value)?;
+                    let mut elem_ll_ty = self.llvm_type(&*elem_ty)?;
+                    let ty_size = self.size_of(elem_ll_ty);
+                    
+                    use self::cmp::GenCmp;
+                    let cmpfunc_ll_fn = self.functions[cmpfunc];
+
+                    let run = context.get_run();
+
+                    // Generate the comparator from the provided custom code.
+                    // The custom comparator can only make use of BinOps that are supported by Weld
+                    // (e.g., lt/gt between scalars, vectors, or structs).
+                    let comparator = self.gen_custom_cmp(&elem_ll_ty,
+                                                         cmpfunc,
+                                                         &cmpfunc_ll_fn)?;
+
+                    // args to qsort_r are: base array pointer, num elements,
+                    // element size, comparator function, run handle
+                    let mut args = vec![elems_ptr, size, ty_size, comparator, run];
+                    let mut arg_tys = vec![LLVMTypeOf(elems_ptr),
+                                           LLVMTypeOf(size),
+                                           LLVMTypeOf(ty_size),
+                                           LLVMTypeOf(comparator),
+                                           LLVMTypeOf(run)];
+                    
+                    // Generate the call to qsort
+                    let void_type = self.void_type();
+                    self.intrinsics.add("qsort_r", void_type, &mut arg_tys); // In-place sort.
```

Comment:
Hmm, this should technically return a new vector, which means we should clone the first one before sorting...unfortunately we don't have code to recursively clone stuff right now, which makes this quite annoying. Maybe add a note about it as a TODO and we can put it in as a patch later on.